From 970e3f27e8e0c21cf8198deb153ecf99ad5aaca1 Mon Sep 17 00:00:00 2001
From: pixelos-gsi <pixelos-gsi@localhost>
Date: Fri, 9 Oct 2026 19:48:51 +0000
Subject: [PATCH 87/87] partial:
 0033-keyguard-fall-back-to-the-primary-bouncer-when-the-s.patch

Change-Id: Ic3fe359345ec52eb41072bd35cc37993b9444f6f
---
 .../android/keyguard/KeyguardAbsKeyInputViewController.java  | 5 +++++
 .../src/com/android/keyguard/KeyguardUpdateMonitor.java      | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardAbsKeyInputViewController.java b/packages/SystemUI/src/com/android/keyguard/KeyguardAbsKeyInputViewController.java
index 127f5da68..6c1e5832b 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardAbsKeyInputViewController.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardAbsKeyInputViewController.java
@@ -240,6 +240,11 @@ public abstract class KeyguardAbsKeyInputViewController<T extends KeyguardAbsKey
         }
     }
 
+    private void abandonSecondFactor() {
+        mKeyguardUpdateMonitor.abandonSecondFactor();
+        getKeyguardSecurityCallback().showCurrentSecurityScreen();
+    }
+
     protected void startErrorAnimation() { /* no-op */ }
 
     protected void verifyPasswordAndUnlock() {
diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java b/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
index 0c6ce8ce5..34ba5e3dd 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
@@ -4113,6 +4113,11 @@ public class KeyguardUpdateMonitor implements TrustManager.TrustListener, CoreSt
             }
             return changed;
         }
+    void abandonSecondFactor() {
+        clearFingerprintRecognized();
+        updateFingerprintListeningState(BIOMETRIC_ACTION_UPDATE);
+    }
+
     }
 
     /**
-- 
2.34.1

